roda-ui/roda-wui/src/main/java/org/roda/wui/client/ingest/process/PluginParameterPanel.java

```diff
@@ -313,6 +317,53 @@ public void onClick(ClickEvent event) {
     removeButton.addStyleName("toolbarLink toolbarLinkSmall");
   }
 
+  /**
+   * Method for creating filterParams for searching aips to dispose: disposaldate has to be prior today and
+   * disposalstatus has to be set in ClientMessages.properties to a relevant value.
+   * dateFinal and preservationstatus_txt are specified search fields in Swedish Customs gip.xml
+   * @return List of filterParams
+   */
+  private static List<FilterParameter> getFilterParameters(){
+    DateTimeFormat dtf = DateTimeFormat.getFormat("yyyy-MM-dd");
+    String dateFromStr="1970-01-01";
+
+    Date dateFrom=dtf.parse(dateFromStr);
+    Date dateTo = new Date();
+
+    DateIntervalFilterParameter dateParam=new DateIntervalFilterParameter("dateFinal","dateFinal",dateFrom,dateTo);
+    List<FilterParameter> parameterList=new ArrayList<>();
+    parameterList.add(new SimpleFilterParameter("preservationstatus_txt",messages.disposalStatus()));
+    parameterList.add(dateParam);
+    return parameterList;
+  }
+
+  /**
+   * Layout corresponding the PluginParameterType.DISPOSAL
+   * The layout displays a preview of which aips that will be disposed when plugin job is run.
+   */
+  private void createDisposalLayout() {
+
+    Label parameterName = new Label(parameter.getName());
+    //pluginParameter button
+    final Button button = new Button(messages.disposalPreviewButton());
```

Comment:
This layout seems pretty generic except for the button message. If existing select AIP can be improved to add this logic it would be much better.